components/org.wso2.carbon.identity.password.history/src/main/java/org/wso2/carbon/identity/password/history/store/Impl/DefaultPasswordHistoryDataStore.java

```diff
@@ -135,6 +143,32 @@ public void remove(User user) throws IdentityPasswordHistoryException {
 
     }
 
+    /**
+     * Delete password history data of a tenant.
+     *
+     * @param tenantId Id of the tenant
+     * @throws IdentityPasswordHistoryException
+     */
+    @Override
+    public void deletePasswordHistoryData(int tenantId) throws IdentityPasswordHistoryException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Deleting all password history data of the tenant: " + tenantId);
+        }
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(PasswordHistoryConstants.SQLQueries
+                        .DELETE_PASSWORD_HISTORY_DATA_BY_TENANT_ID)) {
+                prepStmt.setInt(1, tenantId);
+                prepStmt.execute();
+                IdentityDatabaseUtil.commitTransaction(connection);
+            }
+        } catch (SQLException e) {
+            throw new IdentityPasswordHistoryException(
+                    "Error while deleting password history data of tenant" + tenantId, e);
```

Comment:
Fix the Error message properly. Space should be added at the end. `tenant: `

Reply:
Fixed in #471